### PR TITLE
Pytorch profiler

### PR DESCRIPTION
**Pull request by nicolas-chaulet**

### CODE REVIEW

File: torch_points3d/trainer.py [hunk 1]
```diff
@@ -136,27 +140,44 @@ def _initialize_trainer(self):
     def train(self):
         self._is_training = True
 
-        for epoch in range(self._checkpoint.start_epoch, self._cfg.training.epochs):
-            log.info("EPOCH %i / %i", epoch, self._cfg.training.epochs)
+        with (torch.profiler.profile(
+            schedule=torch.profiler.schedule(
```
Comment (CCInc) [inline torch_points3d/trainer.py:144]: Can we refactor this into its own function to either return a profile or nullcontext?

File: torch_points3d/trainer.py [hunk 2]
```diff
@@ -136,27 +140,44 @@ def _initialize_trainer(self):
...
-                return 0
 
-            if epoch % self.eval_frequency != 0:
-                continue
+            for epoch in range(self._checkpoint.start_epoch, self._cfg.training.epochs):
+                log.info("EPOCH %i / %i", epoch, self._cfg.training.epochs)
 
-            if self._dataset.has_val_loader:
-                self._test_epoch(epoch, "val")
+                with (torch.profiler.record_function('train_epoch') if self.pytorch_profiler_log else nullcontext()):
```
Comment (CCInc) [inline torch_points3d/trainer.py:162]: Maybe refactor this too?

**Review by CCInc — CHANGES_REQUESTED** [review]
Awesome!

File: torch_points3d/trainer.py (hunk 1 above)
Comment (ggrzeczkowicz) [reply torch_points3d/trainer.py:144]: Done with #679

File: torch_points3d/trainer.py (hunk 2 above)
Comment (ggrzeczkowicz) [reply torch_points3d/trainer.py:162]: Done with #679

**Comment by ggrzeczkowicz** [comment]
I refactor torch.profiler.profile and torch.profiler.record_function to use own functions and I move call to profile inside _train_epoch in #679

**Comment by nicolas-chaulet** [comment]
@ggrzeczkowicz looks like static checks are failing, could you please take a look? 
```
make staticchecks
```

**Comment by ggrzeczkowicz** [comment]
I think you could merge that, unless the codecov is a problem.

**Pull request merged**